packages/graphql/src/schemaTypes/objectTypes/gql-RunSpecResponse.ts

```diff
@@ -0,0 +1,30 @@
+import { objectType } from 'nexus'
+import { RunSpecResponseCodeEnum } from '../enumTypes'
+
+export const RunSpecResponse = objectType({
+  name: 'RunSpecResponse',
+  description: 'Result of a runSpec mutation',
+  definition (t) {
+    t.nonNull.field('code', {
+      type: RunSpecResponseCodeEnum,
+    })
+
+    t.string('testingType', {
+      description: 'Testing Type that spec was launched in',
+    })
+
+    t.field('browser', {
+      type: 'Browser',
+      description: 'Browser test was launched in',
+    })
+
+    t.field('spec', {
+      type: 'Spec',
+      description: 'Matched spec that was launched',
+    })
+
+    t.string('detailMessage', {
+      description: 'Contextual information for a given status (typically an expanded error message)',
+    })
+  },
```

Comment:
Did you think about using the pattern of the response being a unioned type?  `RunSpecResponse` could be broken up into something like `RunSpecSuccess` and `RunSpecError` where each type could only contain the fields it needs and mark them as non nullable. Not a must do, but just throwing it out as something to think about.

Reply:
🤔 Hmm, will have to give this some thought. Originally went this direction because my implementation was going to return partial data in error cases (eg: found spec but couldn't launch browser, would send back the matching spec with error code). The way it shaped up there aren't really any useful scenarios for that, so maybe a discriminated return type would be the right way to go.

Reply:
Restructured into Success/Error types